presto-hive/src/main/java/com/facebook/presto/hive/s3select/S3SelectLineRecordReader.java

```diff
@@ -155,6 +173,13 @@ public SelectObjectContentRequest buildSelectObjectRequest(String query, Path pa
         OutputSerialization selectObjectOutputSerialization = buildOutputSerialization();
         selectObjectRequest.setOutputSerialization(selectObjectOutputSerialization);
 
+        if (shouldEnableScanRange()) {
```

Comment:
Do we want to set the scan range parameter if a split entirely covers the entire input file? It seems at best redundant. Seems like we could check `start == 0 && length == HiveFileSplit#getFileSize()`  to skip this.

Reply:
It is not impacting the request and the HiveFileSplit is not available in this context.

Reply:
Passed the fileSize and added the check